test/unit/preprocessing/test_preprocessing_through_api.py

```diff
@@ -195,8 +198,9 @@ def data_with_text_features_and_nans():
 def test_correct_api_dataset_preprocessing():
     """ Check if dataset preprocessing was performed correctly when API launch using. """
     funcs = [data_with_only_categorical_features, data_with_too_much_nans,
-             data_with_spaces_and_nans_in_features, data_with_nans_in_target_column,
-             data_with_nans_in_multi_target]
+             data_with_spaces_and_nans_in_features, data_with_nans_in_target_column,]
+             # TODO: how deal with multi-target in xgboost and lightgbm?
```

Comment:
Не очень понятен вопрос - как и везде.

Reply:
Наверное, правильнее было бы создать issue, а не PR. 

В конкретно этом месте не предлагается какого-либо решения. Здесь комментарий был добавлен, чтобы локализовать случай, когда в начальном пайплайне присутствуют xgboost или lgbm и стоит задача с мульти-таргетом. Тест не проходит, потому что, при конвертации класса `input_data` в датафрейм происходит развертывание массива с таргетом (т.е., к примеру, в обычной задаче был массив с таргетом размером 3х1, а в задаче с мульти-таргетом он имеет размер 3х2, поэтому после вызова `np.ravel()` в методе `convert_to_dataframe` он преобразуется в 1-d массив с длиной 3х2=6, следовательно, появится ошибка, связанная с длиной массива). 
```python
# вырезка из метода convert_to_dataframe
        if copied_input_data.target is not None and copied_input_data.target.size > 0:
            rows_len = dataframe.shape[0]
            target = copied_input_data.target[:rows_len]
            dataframe['target'] = np.ravel(target)
```
Этот случай я сейчас пытаюсь обработать.

---

Касаемо этого вопроса №3: `FAILED test #0 contains class label "0" that is not present in the learn dataset`. 
В тесте в качестве датасета с признаками выступает небольшая таблица `features`. При разбивке её на train и eval внутри FEDOT через `train_test_data_setup` упускается часть классов, поэтому, после обучения мы встречаем неизвестный до этого класс "0".

```python
def data_with_categorical_target(with_nan: bool = False):
    """
    Generate dataset for classification task where target column is defined as
    string categories (e.g. 'red', 'green'). Dataset is generated so that when
    split into training and test in the test sample in the target will always
    be a new category.

    :param with_nan: is there a need to generate target column with np.nan
    """
    task = Task(TaskTypesEnum.classification)
    features = np.array([[0, 0],
                         [0, 1],
                         [8, 8],
                         [8, 9]])
    if with_nan:
        target = np.array(['blue', np.nan, np.nan, 'di'], dtype=object)
    else:
        target = np.array(['blue', 'da', 'ba', 'di'], dtype=str)
    train_input = InputData(idx=np.array([0, 1, 2, 3]), features=features,
                            target=target, task=task, data_type=DataTypesEnum.table,
                            supplementary_data=SupplementaryData())

    return train_input
```
```python
            train_input, eval_input = train_test_data_setup(input_data)

            X_train, y_train = self.convert_to_dataframe(
                train_input, identify_cats=self.params.get('enable_categorical')
            )

            X_eval, y_eval = self.convert_to_dataframe(
                eval_input, identify_cats=self.params.get('enable_categorical')
            )
```

Предлагаемое решение: удвоить в тесте `test_categorical_target_processed_correctly` в методе получения данных  `data_with_categorical_target` массивы `features` и `target`. В таком случае ошибка не появляется и тест проходит. 
![image](https://github.com/user-attachments/assets/3a0e3015-5b49-4d4b-bf48-5c7c1e350f58)

---

Касаемо вопроса №4: `FAILED ValueError: Length of values (5) does not match length of index (6)`.
В тесте массив `features` имеет количество строк = 6, а длина массива `target` = 5.
![image](https://github.com/user-attachments/assets/661c83c0-bfb3-43a4-96b1-a09fdf5b5551)
Предлагаемое решение: укоротить `features` на 1. Если так сделать - тест проходит, ошибка не появляется.